### DISCUSSION TITLE
Error in 02-QbC for water sytem

### DISCUSSION THREAD

**Discussion by sahideep** [MarsalekGroup/aml#29] [Q&A] [answered]
Dear AML team,

I am trying to train pure water system using AML following the steps according to my system mentioned in your git-hub repository.  I am facing problems in 02 -QbC step as it is unable to find nnp-scaling command. I dont't have access to /bin/sh path, as it is clearly can be seen that it is looking for nnp-scaling command in /bin/sh. Is there any way, I can export or change the path of n2p2 in aml?

![Screenshot from 2023-11-20 15-05-39](https://github.com/MarsalekGroup/aml/assets/141631470/58bd08ed-8b7c-4075-978c-74347cd160d1)
![Screenshot from 2023-11-20 15-00-18](https://github.com/MarsalekGroup/aml/assets/141631470/d001962f-09db-4149-be1b-09649c497e4a)

Thank you

**Comment by OndrejMarsalek** [comment, marked as answer]
This means that the `nnp-scaling` executable could not be found in your executable path when AML tried to launch it. It is reported by `/bin/sh` because AML runs these commands in a shell, so it launches `/bin/sh` which then tries to launch `nnp-scaling`. Here, that step failed. The failure reported in the Python traceback shows that AML was not able to read the output. Better error reporting is eventually coming to the released version of the code.

You can address this by checking your environment and making sure `nnp-scaling` (and other `nnp-*` programs) can be launched in the same environment that AML is running in.
